pytorch_lightning/callbacks/pt_callbacks.py

```diff
@@ -21,6 +21,7 @@ class Callback(object):
     def __init__(self):
         self.validation_data = None
         self.model = None
+        self.trainer = None
```

Comment:
I would move all these as protected, so `self._model`, `self._trainer`, etc.

Reply:
Done. Moreover, I have removed things like `.set_model()` and `.set_params()` because it was not really used anywhere.